smarts/sstudio/genscenario.py

```diff
@@ -145,6 +141,67 @@ def gen_traffic(
         logger.debug(f"Generated traffic for scenario={scenario}")
 
 
+def gen_social_agents(
+    scenario: str,
+    social_actor_mission_pairs: Sequence[Tuple[types.SocialAgentActor, types.Mission]],
+    name: str,
+    seed: int = 42,
+    overwrite: bool = False,
+):
+    """Generates the social agent missions for the given scenario.
+
+    Args:
+        scenario:
+            The scenario directory
+        missions:
+            A sequence of missions for social agents to perform
+        social_agent_actor(s):
+            The actor(s) to use
+        name:
+            A short name for this grouping of social agents. Is also used as the name
+            of the social agent traffic file
+        seed:
+            The random seed to use when generating behaviour
+        overwrite:
+            If to forcefully write over the previous existing output file
+    """
+
+    # For backwards compatibility we support both a single value and a sequence
+    social_actor_mission_pairs = social_actor_mission_pairs
+    if not isinstance(social_actor_mission_pairs, collections.abc.Sequence):
+        raise ValueError(f"{list(social_actor_mission_pairs)}")
+
+    ActorMissionPairs = Sequence[Tuple[types.SocialAgentActor, types.Mission]]
+    actors: ActorMissionPairs = [a for a, _ in social_actor_mission_pairs]
+    missions: Sequence[types.Mission] = [m for _, m in social_actor_mission_pairs]
+    # This doesn't support BoidAgentActor. Here we make that explicit
+    if any(isinstance(actor, types.BoidAgentActor) for actor in actors):
+        raise ValueError(
+            f"{gen_social_agents.__name__}(...) can't be called with BoidAgentActor, got:"
+            f"{social_actor_mission_pairs}"
+        )
+
+    actor_names = [a.name for a in actors]
+    if len(actor_names) != len(set(actor_names)):
+        raise ValueError(f"Actor names={actor_names} must not contain duplicates")
+    if len(missions) != len(set(missions)):
+        raise ValueError(f"Supplied Missions must be unique.")
```

Comment:
Why does `(actor, mission)` pair needs to be unique? Why can't the same actor have different missions or vice-versa?

Reply:
Doesn't this seems to contradict the `(actor, mission)` we create in `4_lane_t` map where the two actors have the same `reused_mission`?

Reply:
Good catch, it looks like this is not getting triggered properly.

Reply:
Actually, it looks like this is triggered properly but the terminology was not clear. I have updated `Scenario.social_agents` -> `Scenario.social_agent_groups` to remove the confusion.